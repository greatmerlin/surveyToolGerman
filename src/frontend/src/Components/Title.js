import React from "react";

/**
 * Renders the title of the page.
 *
 * @param {children} JSX.Element
 * @returns JSX.Element
 */

export default function Title({ children }) {
  return <h2 className="mb-4">{children}</h2>;
}
