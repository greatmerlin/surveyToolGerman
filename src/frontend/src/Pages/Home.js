import React from "react";
import Title from "../Components/Title";
import { Button } from "react-bootstrap";

/**
 * Homepage.
 *
 * @returns JSX.Element
 */

export default function Home() {
  const introList = [
    "Erstelle einen Fragebogen mit einem benutzerfreundlichen und einfach verständlichen Umfragetool.",
    "Einfach verständliche Analyse von Umfrageergebnissen.",
    "Answerium, da oft Fragen auftauchen!",
  ];

  return (
    <>
      <Title>Online Umfragen</Title>
      <ul style={{ marginTop: "25vh" }}>
        {introList.map((entry, i) => {
          return (
            <li className="my-3" key={i}>
              <p className="lead">{entry}</p>
            </li>
          );
        })}
      </ul>
      <Button
        href="/selectPath"
        variant="primary"
        className="my-3"
        data-testid="testSelectPath"
      >
        Jetzt kostenlos starten!
      </Button>
    </>
  );
}
