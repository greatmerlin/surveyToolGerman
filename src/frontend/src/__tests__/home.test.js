import Home from "../Pages/Home.js";
import React from "react";
import { render } from "@testing-library/react";
import Title from "../Components/Title.js";
import { Button } from "react-bootstrap";
import { create } from "react-test-renderer";
import "@testing-library/jest-dom";

// tests that the <Home /> component renders without any problem (describe) 
describe("My first snapshot test", () => {
  test("testing app button", () => {
    let tree = create(<Home />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

// test a custom Home Component, if it renders without any problem -> everything renders (test)
test("<Home /> matches snapshot", () => {
  const component = render(
    <>
      <Title>Online Umfragen</Title>
      <ul style={{ marginTop: "25vh" }}>
        {[
          "Erstelle einen Fragebogen mit einem benutzerfreundlichen und einfach verständlichen Umfragetool.",
          "Einfach verständliche Analyse von Umfrageergebnissen.",
          "Answerium, da oft Fragen auftauchen!",
        ].map((entry, i) => {
          return (
            <li className="my-3" key={i}>
              <p className="lead">{entry}</p>
            </li>
          );
        })}
      </ul>
      <Button href="/selectPath" variant="primary" className="my-3">
        Jetzt kostenlos starten!
      </Button>
    </>
  );

  expect(component.container).toMatchSnapshot();
});

// now let's test if the component renders without crashing
// we do the same thing as react, we pass the elements into a div and render it
// if a component doesn't render, then the test will fail and we know that our code crashes
test("<Home /> renders without crashing", () => {
  const div = document.createElement("div");
  render(
    <>
      <Title>Online Umfragen Mocjup</Title>
      <ul>
        {["one", "two"].map((entry, i) => {
          return (
            <li className="my-3" key={i}>
              <p className="lead">{entry}</p>
            </li>
          );
        })}
      </ul>
      <Button href="/selectPath" variant="primary" className="my-3">
        Jetzt kostenlos starten!
      </Button>
    </>,
    div
  );
});
