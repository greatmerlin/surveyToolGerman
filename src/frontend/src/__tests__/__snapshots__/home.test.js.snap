// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> matches snapshot 1`] = `
<div>
  <h2
    class="mb-4"
  >
    Online Umfragen
  </h2>
  <ul
    style="margin-top: 25vh;"
  >
    <li
      class="my-3"
    >
      <p
        class="lead"
      >
        Erstelle einen Fragebogen mit einem benutzerfreundlichen und einfach verst채ndlichen Umfragetool.
      </p>
    </li>
    <li
      class="my-3"
    >
      <p
        class="lead"
      >
        Einfach verst채ndliche Analyse von Umfrageergebnissen.
      </p>
    </li>
    <li
      class="my-3"
    >
      <p
        class="lead"
      >
        Answerium, da oft Fragen auftauchen!
      </p>
    </li>
  </ul>
  <a
    class="my-3 btn btn-primary"
    href="/selectPath"
  >
    Jetzt kostenlos starten!
  </a>
</div>
`;

exports[`My first snapshot test testing app button 1`] = `
Array [
  <h2
    className="mb-4"
  >
    Online Umfragen
  </h2>,
  <ul
    style={
      Object {
        "marginTop": "25vh",
      }
    }
  >
    <li
      className="my-3"
    >
      <p
        className="lead"
      >
        Erstelle einen Fragebogen mit einem benutzerfreundlichen und einfach verst채ndlichen Umfragetool.
      </p>
    </li>
    <li
      className="my-3"
    >
      <p
        className="lead"
      >
        Einfach verst채ndliche Analyse von Umfrageergebnissen.
      </p>
    </li>
    <li
      className="my-3"
    >
      <p
        className="lead"
      >
        Answerium, da oft Fragen auftauchen!
      </p>
    </li>
  </ul>,
  <a
    className="my-3 btn btn-primary"
    data-testid="testSelectPath"
    href="/selectPath"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    Jetzt kostenlos starten!
  </a>,
]
`;
